mirrord/kube/src/api/container.rs

```diff
@@ -125,18 +126,33 @@ pub fn check_mesh_vendor(pod: &Pod) -> Option<MeshVendor> {
     }
 
     let container_statuses = pod.status.as_ref()?.container_statuses.as_ref()?;
+    let container_names = container_statuses
+        .iter()
+        .chain(
+            pod.status
+                .as_ref()?
+                .init_container_statuses
+                .as_ref()?
+                .iter(),
+        )
+        .map(|status| status.name.as_str())
+        .collect::<Vec<&str>>();
+
+    // check that all the containers are present
+    // we had a case where istio cni was detected as istio while
+    // the init was only present.
+    // leave old logic for detection to not break existing setups
+    if ISTIO_CNI.iter().all(|name| container_names.contains(name)) {
```

Comment:
Is there a reason for contains? like are the container names different from the lookup? (container named `my-custom-istio-validation` will pass this filter) I'd maybe do a straight up compare

Reply:
Nice catch!

Reply:
Ah sorry, `container_names` is a Vec of names

Reply:
meaning, it does exact match